### DISCUSSION TITLE
Why does Microsoft.OpenJDK.17 manifiest include UpgradeBehavior: uninstallPrevious?

### DISCUSSION THREAD

**Discussion by ave9858** [microsoft/winget-pkgs#109267] [Q&A]
I noticed this while upgrading my Java through winget. My features I set on the install (enable JAVA_HOME) were not preserved during the upgrade, and I had to reinstall manually to change the installed features. When I tested the installer manually in a VM, I noticed it automatically removed older versions of Microsoft OpenJDK 17, so I'm not sure why the manifests tell winget to uninstall older versions when the installer seems to do that automatically and in a way that preserved installed features.

**Comment by superusercode** [comment]
It looks like that was introduced with the second to oldest manifest without any reason describing it, and I don't know if it was intentional for that to be done.

**Comment by ave9858** [comment]
Seems only some of the newer versions can be automatically uninstalled and replaced with newer versions. Since the UpgradeBehavior applies to all previous versions, it has to be set to uninstallPrevious in case the current version of openjdk is very old.
